### DISCUSSION TITLE
How do I define the endpoints?

### DISCUSSION THREAD

**Discussion by Mpellet771** [trevor-viljoen/mlbapi#5] [Q&A] [answered]
I think I understand the endpoints, but this command:

game_list = mlbapi.game([parameters])

Gets me this error:

game_list = mlbapi.game('ALS')
AttributeError: module 'mlbapi' has no attribute 'game'

mlbapi.schedule works but no other endpoints. What am I missing?

**Comment by trevor-viljoen** [comment, marked as answer]
Hello and thanks for using the code. It looks like you want a list of games. I'm not sure what "ALS" refers to in your code sample, but here is an example of how to use the schedule endpoint to get a list of regular season games for a given team in a specific season.

```
import mlbapi

astros_schedule = mlbapi.schedule(season="2023", game_types="R", team_id=117)
```

This gives back a list of regular season games for the 2023 season for the Houston Astros.

From that list of games, you can get information about individual games such as the `game_pk` that is needed for checking a `BoxScore`.

```
game_pk = astros_schedule.dates[0].games[0].game_pk
box_score = mlbapi.boxscore(game_pk)
```

This gets the boxscore, using the game primary key for the first game of the first date of the schedule.

If you have an example of something you're wanting to do, I can help with that. I will be pushing an update that you will need in order to follow the example above. Some breaking changes have occurred with the API since I last pushed a commit.

**Reply by Mpellet771** [reply]
Thank you!

**Reply by Mpellet771** [reply]
OK, I am trying to get all of the games for 2023 so far. Here is what I have in my code:import mlbapi

astros_schedule = mlbapi.schedule(season="2023", game_types="R")

game_pk = astros_schedule.dates.games.game_pk

print(game_pk)

box_score = mlbapi.boxscore(game_pk)


Here is the response:

https://statsapi.mlb.com/api/v1/schedule {'season': '2023', 'gameTypes': 'R', 'sportId': 1}
Traceback (most recent call last):
  File "C:/Users/maria/Documents/Baseball/Python/ScraperMLBapi.py", line 5, in <module>
    game_pk = astros_schedule.dates.games.game_pk
AttributeError: 'list' object has no attribute 'games'

I think I need to use a loop to catch all of them (your code got me one game_pk), but am wondering about the attribute error. Thanks

**Reply by trevor-viljoen** [reply]
The API returns games as a list for each date. Most times there will only be a single game for any given day, but there could be a double header. So you will need to loop through the dates and the games lists.

```
for day in astros_schedule.dates:
    for game in day.games:
        game_pk = game.game_pk
        box_score = mlbapi.boxscore(game_pk)
```

**Comment by Mpellet771** [comment]
Thank you! I’ll try it when I return from my trip.Marianne Sent from my iPhoneOn Jul 17, 2023, at 9:53 AM, Trevor Viljoen ***@***.***> wrote:﻿
The API returns games as a list for each date. Most times there will only be a single game for any given day, but there could be a double header. So you will need to loop through the dates and the games lists.
for day in astros_schedule.dates:
    for game in day.games:
        game_pk = game.game_pk
        box_score = mlbapi.boxscore(game_pk)
